dev/TabView/TabView.cpp

```diff
@@ -350,14 +350,49 @@ void TabView::OnScrollViewerLoaded(const winrt::IInspectable&, const winrt::Rout
 
         auto increaseButton = SharedHelpers::FindInVisualTreeByName(scrollViewer, L"ScrollIncreaseButton").as<winrt::RepeatButton>();
         m_scrollIncreaseClickRevoker = increaseButton.Click(winrt::auto_revoke, { this, &TabView::OnScrollIncreaseClick });
+
+        m_scrollViewerViewChangedRevoker = scrollViewer.ViewChanged(winrt::auto_revoke, { this, &TabView::OnScrollViewerViewChanged });
     }
 
     UpdateTabWidths();
 }
 
+void TabView::OnScrollViewerViewChanged(winrt::IInspectable const& sender, winrt::ScrollViewerViewChangedEventArgs const& args)
+{
+    UpdateScrollViewerDecreaseAndIncreaseButtonsViewState();
+}
+
+void TabView::UpdateScrollViewerDecreaseAndIncreaseButtonsViewState()
+{
+    if (auto&& scrollViewer = m_scrollViewer.get())
+    {
+        auto decreaseButton = SharedHelpers::FindInVisualTreeByName(scrollViewer, L"ScrollDecreaseButton").as<winrt::RepeatButton>();
+        auto increaseButton = SharedHelpers::FindInVisualTreeByName(scrollViewer, L"ScrollIncreaseButton").as<winrt::RepeatButton>();
+
+        auto minThreshold = 0.1;
+
+        if (abs(scrollViewer.HorizontalOffset() - scrollViewer.ScrollableWidth()) < minThreshold)
+        {
+            decreaseButton.IsEnabled(true);
+            increaseButton.IsEnabled(false);
```

Comment:
I think better than setting the properties from code behind would be to add visual states to the template and calling go to state

Reply:
I am not sure if using VisualState is the way to go as I am not seeing much benefits here. We still need to do some calculation here to disable or enable it on the fly based on min threshold since it is a floating value. (Or we will have this magic number lives inside xaml?) On the other hand we need to make sure it works in all cases like Window resizing. Maybe we should rewrite this whole scrolling buttons thingy in TabView vNext? @chingucoding

Reply:
I am not sure if with TabView vNext, a rewrite is a) a good idea and b) a rewrite would fix this at all. After all, we need to provide some sort of scrolling, and disabling those buttons is something we have to do. @stmoy FYI